discovery_engine_core/web-api/src/models.rs

```diff
@@ -96,6 +105,20 @@ impl DocumentId {
 #[derive(Clone, Debug, Display, Serialize, Deserialize, PartialEq, Eq, Hash)]
 pub struct DocumentPropertyId(String);
 
+impl DocumentPropertyId {
+    pub(crate) fn new(id: impl Into<String>) -> Result<Self, Error> {
```

Comment:
:pushpin:  See the other comment I mentioned but we probably want to limit the format of property ids to e.g. `[a-A0-9_]+` or similar.